### PR TITLE
Reduce precision in select tests to pass on alternate architectures

### LINKED ISSUE #1174: Builds/Tests now failing on aarch64, ppc64le architectures (Fedora 36=Rawhide)
**Problem description**

All builds on the ppc64le architecture with F36/Rawhide now fail. This was not the case four days ago (see https://copr.fedorainfracloud.org/coprs/fuller/Cantera/builds/).

Concurrently, the `kinetics: KineticsAddSpecies3.add_species_sequential` test now fails on "successful" builds on F34/35 for ppc64le architectures (and also for aarch64, i686 and s390x on all three Fedoras 34, 35, and 36/Rawhide, but not x86_64). This may not be a new problem as I only just added testing automation to the build automation for all architectures. (Yes, I know, I should have done that earlier)

**Steps to reproduce**

`scons build && scons test`

**Behavior**

Error message:
Build failure (ppc64le:F36):
```
In file included from /usr/include/eigen3/Eigen/Core:210,
                 from /usr/include/eigen3/Eigen/SparseCore:11,
                 from /usr/include/eigen3/Eigen/Sparse:26,
                 from include/cantera/numerics/eigen_sparse.h:10,
                 from include/cantera/kinetics/StoichManager.h:11,
                 from include/cantera/kinetics/Kinetics.h:14,
                 from src/base/Solution.cpp:12:
/usr/include/eigen3/Eigen/src/Core/arch/AltiVec/PacketMath.h:78:8: internal compiler error: Segmentation fault
   78 | static _EIGEN_DECLARE_CONST_FAST_Packet4i(ZERO, 0); //{ 0, 0, 0, 0,}
      |        ^
```
Test failure (all other affected systems):
```
[----------] 2 tests from KineticsAddSpecies3
[ RUN      ] KineticsAddSpecies3.add_species_sequential
test/kinetics/kineticsFromScratch3.cpp:371: Failure
Expected equality of these values:
  k_ref[i]
    Which is: 591054161.41004908
  k[i]
    Which is: 591054161.41004813
i = 0; N = 4
test/kinetics/kineticsFromScratch3.cpp:377: Failure
Expected equality of these values:
  k_ref[i]
    Which is: 150.5822178080069
  k[i]
    Which is: 150.58221780800667
i = 0; N = 4
test/kinetics/kineticsFromScratch3.cpp:384: Failure
Expected equality of these values:
  w_ref[iref]
    Which is: 150.58221780800866
  w[i]
    Which is: 150.58221780800844
sp = O; N = 4
[  FAILED  ] KineticsAddSpecies3.add_species_sequential (5 ms)
```

**System information**

- Cantera version: fcff5929225a0728dce33e44acf21149f6fb928f
- OS: Fedora Linux 36 (Rawhide), Fedora 35, Fedora 34
- Python/MATLAB/other software versions: Python 3.9, 3.10 (see logs)

**Attachments**

**Additional context**
While the build processes are not failing, one test pertaining to kinetics is on both F34 and F35.
I suspect that these problems are related.
The test failures looks like excessive precision being requested - or is this a truncation/rounding error?

**Logs**
[ppc64le/Rawhide - failed build](https://download.copr.fedorainfracloud.org/results/fuller/Cantera/fedora-rawhide-ppc64le/03192999-cantera/builder-live.log.gz)
[ppc64le/F35 - failed test](https://download.copr.fedorainfracloud.org/results/fuller/Cantera/fedora-35-ppc64le/03192999-cantera/builder-live.log.gz)
[ppc64le/F34 - failed test](https://download.copr.fedorainfracloud.org/results/fuller/Cantera/fedora-34-ppc64le/03192999-cantera/builder-live.log.gz)

[aarch64/Rawhide](https://download.copr.fedorainfracloud.org/results/fuller/Cantera/fedora-rawhide-aarch64/03192999-cantera/builder-live.log.gz)
[aarch64/F35](https://download.copr.fedorainfracloud.org/results/fuller/Cantera/fedora-35-ppc64le/03192999-cantera/builder-live.log.gz)
[aarch64/F34](https://download.copr.fedorainfracloud.org/results/fuller/Cantera/fedora-34-aarch64/03192999-cantera/builder-live.log.gz)

Additional information and build logs at:
1) https://copr.fedorainfracloud.org/coprs/fuller/Cantera/build/3192999/
2) https://koji.fedoraproject.org/koji/taskinfo?taskID=81397194

### PR DESCRIPTION
**Pull request by mefuller**
**Changes proposed in this pull request**
Following on the discussion in #1174, minor changes in test tolerances are introduced to allow tests to pass on non-x86_64 architectures
- Replace `EQ` with `NEAR` on several kinetics tests which were affecting all alternate architectures (https://copr.fedorainfracloud.org/coprs/fuller/Cantera/build/3192999/)
- Increase test tolerances from 1e-9 to 2e-9 in Chebyshev tests for which 32-bit i386 systems had errors of 1.6e-9 (https://copr.fedorainfracloud.org/coprs/fuller/cantera-test/build/3195208/)

**If applicable, fill in the issue number this pull request is fixing**
This closes #1174 as it corrects test/build failures with the following exceptions:
Test failures for s390x architecture and build failure on ppc64le (F36/Rawhide only) are identified there as originating outside of Cantera and will be addressed upstream.

**Checklist**

- [X] The pull request includes a clear description of this code change
- [X] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [X] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

**Comment by mefuller** [comment]
Alternate arch builds succeed for Fedora Rawhide: https://koji.fedoraproject.org/koji/taskinfo?taskID=81443230
Alternate arch builds succeed across Fedora releases: https://copr.fedorainfracloud.org/coprs/fuller/cantera-test/build/3197399/

File: test/kinetics/kineticsFromScratch3.cpp [hunk 1]
```diff
@@ -368,20 +368,20 @@ class KineticsAddSpecies3 : public testing::Test
...
-            EXPECT_DOUBLE_EQ(k_ref[i], k[i]) << "i = " << i << "; N = " << N;
+            EXPECT_NEAR(k_ref[i], k[i], k_ref[i]*1e-12) << "i = " << i << "; N = " << N;
         }
 
         kin.getCreationRates(w.data());
         kin_ref->getCreationRates(w_ref.data());
         for (size_t i = 0; i < kin.nTotalSpecies(); i++) {
             size_t iref = p_ref.speciesIndex(p.speciesName(i));
-            EXPECT_DOUBLE_EQ(w_ref[iref], w[i]) << "sp = " << p.speciesName(i) << "; N = " << N;
+            EXPECT_NEAR(w_ref[iref], w[i], w_ref[i]*1e-12) << "sp = " << p.speciesName(i) << "; N = " << N;
```
Comment (speth) [inline test/kinetics/kineticsFromScratch3.cpp]: The tolerance should be `w_ref[iref]*1e-12`, right?

**Review by speth — CHANGES_REQUESTED** [review]
These changes look reasonable to me, with one small fix.

File: test/kinetics/kineticsFromScratch3.cpp (hunk 1 above)
Comment (mefuller) [reply test/kinetics/kineticsFromScratch3.cpp]: yup - my mistake

**Review by speth — APPROVED** [review]

**Pull request merged**